ts/Core/Tooltip.ts

```diff
@@ -1134,6 +1133,20 @@ class Tooltip {
                                 Palette.neutralColor60
                             )
                         });
+
+                    } else if (currentSeries.options.className) {
+
+                        this.chart.series.forEach(
+                            function (s): void {
+                                label.removeClass(
+                                    s.options.className as string
+                                );
+                            }
+                        );
+
+                        label.addClass(
+                            currentSeries.options.className as string
+                        );
```

Comment:
This code block can be removed now, right?

Reply:
Unfortunately if we remove it we will loose the `className`-colored border of the tooltip. I have assumed that tooltips border should be colored by series, regardless of custom point color, however I am not sure if this is appropriate.

Reply:
Okay, let's simplify this. We can create a function `Tooltip.prototype.getClassName` that always delivers the full, appropriate class name for the label, then we don't need to deal with adding, removing, regexing etc. at all. See [Point.prototype.getClassName](https://github.com/highcharts/highcharts/blob/3776e7a402618302a6ac5cc534edfa69cde9096c/ts/Core/Series/Point.ts#L562-L582) for inspiration. Remember to also use it in the Tooltip `getLabel` function (where point and series are undefined I guess).

Reply:
PS: actually it may be that we don't need to set a class name at all in `getLabel`, because it always passes `refresh` before it is actually displayed? You can check that out.

Reply:
I will!

There is some code in the `getLabel`-function which takes any classes supplied in `tooltip.options.className` (can be seen at [line 446](https://github.com/highcharts/highcharts/blob/3776e7a402618302a6ac5cc534edfa69cde9096c/ts/Core/Tooltip.ts#L446)) and includes them in the creation of the label, last arg in the assignment at [line 522](https://github.com/highcharts/highcharts/blob/3776e7a402618302a6ac5cc534edfa69cde9096c/ts/Core/Tooltip.ts#L522)

Should we move that code to the new `getClassName`-function?

Reply:
Yes, the new `getClassName` should add the tooltip class name, the series class name and the point class name in succession. That should work 🤞